synapse/storage/registration.py

```diff
@@ -92,6 +92,13 @@ def get_user_by_id(self, user_id):
             query, user_id
         )
 
+    def get_all_users(self):
+        query = ("SELECT users.name FROM users")
+        return self._execute(
+            self.cursor_to_dict,
+            query
+        )
```

Comment:
A mild annoyance with using `_execute` is that it doesn't name the transaction nicely, so we can't get nice database transaction performance data. Might be worth doing one of: a) threading through a name, b) using `runInteraction` directly.